beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -3833,6 +3890,10 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         // end up with blocks in fork choice that are missing from disk.
         // See https://github.com/sigp/lighthouse/issues/2028
         let (_, signed_block, blobs, data_columns) = signed_block.deconstruct();
+        let custody_columns_count = self.data_availability_checker.get_custody_columns_count();
+        let data_columns = data_columns
+            .filter(|a| a.len() >= custody_columns_count)
+            .or_else(|| data_column_recv?.blocking_recv());
```

Comment:
Add a nice comment here explaining the mechanism of `data_column_recv` and why trigger it here under these conditions